testapi.pm

```diff
@@ -164,19 +164,25 @@ sub save_screenshot {
 
 =head2 record_soft_failure
 
-  record_soft_failure([$reason]);
+=for stopwords softfail
+
+  record_soft_failure([$reason] [, force_status => $force_status]);
 
 Record a soft failure on the current test modules result. The result will
 still be counted as a success. Use this to mark where workarounds are applied.
-Takes an optional C<$reason> string which is recorded in the log file.
+Takes an optional C<$reason> string which is recorded in the log file.  Set
+C<$force_status> to a true value to override the test module status to a
+softfail status even if the module would be set to fail otherwise. This can be
```

Comment:
I find the use of this flag a little undermotivated here. Because what you write here makes *only* sense in post_fail_hooks, actually the whole force_status does. Because you need to call it *after* the failure - which can't be in run() as failures are exceptions throwing you out of the run.

And IMO this should be an extra API call - and not interfered into record_soft_failure.

Reply:
I see. We already have `record_info` with various options but to my understanding it *only* informs and does not change the status. So how about a third explicit API call `record_known_failure`?

Reply:
We're getting into troubled water here - if a soft fail is defined as known failure, what's the difference between record_soft_failure and record_known_failure? None to the reader.

So I wouldn't introduce this other word for the same - but make the reader aware we're doing more than recording in the post fail hook: force_soft_failure

Reply:
sure, works for me, updated